### PR TITLE
upgrade dependencies to prepare jdk17 upgrade

### PR DESCRIPTION
**Pull request by stevie9868**
**Background:**
We are not able to upgrade to jdk17 right now since hms is too old and is not compatible with jdk17. We will deprecate hms soon, and once it is deprecated, metacat can be upgraded to jdk17.
For this PR, we update metacat dependencies besides hms to be compatible with jdk17. 

**Testing plan:**
1. JDK8 Build pass
2. Run ./gradlew clean build under jdk17, and build succeed with all unit tests pass.
3. But ./gradlew functionalTest for jdk17 failed due to hms incompatability issue.

### CODE REVIEW

File: codequality/findbugs/excludeFilter.xml [hunk 1]
```diff
@@ -25,4 +25,8 @@
         <Bug code="RCN" />
     </Match>
 
+    <Match>
+        <Bug pattern="CT_CONSTRUCTOR_THROW"/>
```
Comment (rsyncv) [inline codequality/findbugs/excludeFilter.xml:29]: Good, excluding this is the right way to go

**Review by rsyncv — COMMENTED** [review]
Could you add a description for why the hms incompatability issue still appears in functional tests even if the java version for the whole project is not updated?

**Comment by stevie9868** [comment]
When we try to run functional test under jdk 17, we get the below exception:
Caused by: javax.jdo.JDOFatalInternalException: The java type java.lang.Integer (jdbc-type="", sql-type="") cant be mapped for this datastore. No mapping is available.

Doing some search online:
https://issues.apache.org/jira/browse/SPARK-27538, hms is has a dataneucleus dependency which is not compatible with jdk17. 
Also looks like the process to migrate hms to be compatible jdk11 is still not complete https://issues.apache.org/jira/browse/HIVE-22415

Thus we are not able to upgrade metacat to jdk17 due to hms incompatibility with java version older than 8.

File: metacat-connector-s3/build.gradle [hunk 2]
```diff
@@ -21,16 +21,16 @@ dependencies {
...
-    compile("com.google.inject.extensions:guice-multibindings")
-    compile("com.google.inject.extensions:guice-servlet")
-    compile("joda-time:joda-time")
-    compile("org.hibernate:hibernate-entitymanager")
+    implementation("com.google.guava:guava")
+    implementation("com.google.inject:guice")
+    implementation("com.google.inject.extensions:guice-persist")
+    implementation("com.google.inject.extensions:guice-multibindings")
+    implementation("com.google.inject.extensions:guice-servlet")
+    implementation("joda-time:joda-time:2.10.10")
```
[the remark this answers is not in the document]
Comment (stevie9868) [reply metacat-connector-s3/build.gradle:32]: If I don't add the version, I will get the following error.

 Failed to resolve 'joda-time:joda-time' for project 'metacat-connector-s3'
  The following dependencies are missing a version: joda-time:joda-time

**Review by rsyncv — APPROVED** [review]
Thanks for adding more context, good work

**Pull request merged**